source/core_json.c

```diff
@@ -1028,6 +1029,12 @@ static bool skipObjectScalars( const char * buf,
         }
     }
 
+    /* An empty JSON object is valid. */
+    if( i == origStart )
```

Comment:
Not sure if my understanding is correct.
The first check in the loop is skipString check. skipString check moves index i only when success. It returns false with the following conditions:
* pairing '"' not found
* '\\' not followed by an escaped-character
* control character
* non-UTF8 characters

Will it return true in with these errors?
Example json file:
```
{
    "foo": { " }
}
```

Reply:
That is a good catch. Changed the detection of empty array and empty JSON object in value as per our discussion.